### PR TITLE
Remove retyped and renamed storage layout annotations

### PR DESCRIPTION
**Pull request by frangio**
Removes all `@custom:oz-retyped-from` and `@custom:oz-renamed-from` annotations for storage layout checks from OpenZeppelin Upgrades Plugins.

Also removes the `@custom:storage-size` annotations for the transpiler, all of them found in `upgradeable.patch`.

The changes to storage layout need to be documented in the changelog but the storage-size change is the least of changes given the upcoming change to namespaced storage.

Fixes LIB-932

### CODE REVIEW

File: scripts/upgradeable/upgradeable.patch [hunk 1]
```diff
@@ -202,12 +137,12 @@ index d94e956a..86bb5713 100644
...
  
  /**
-@@ -30,27 +29,19 @@ import {IERC5267} from "../../interfaces/IERC5267.sol";
+@@ -29,28 +28,18 @@ import {IERC5267} from "../../interfaces/IERC5267.sol";
+  * separator from the immutable values, which is cheaper than accessing a cached version in cold storage.
   *
   * _Available since v3.4._
-  *
+- *
 - * @custom:oz-upgrades-unsafe-allow state-variable-immutable state-variable-assignment
```
Comment (Amxx) [inline scripts/upgradeable/upgradeable.patch:145]: The patch removes this from the vanilla version ... but if we don't want it in the upgradeable version, why is it in the vanilla version in the first place ?

Comment (ernestognw) [reply scripts/upgradeable/upgradeable.patch:145]: I think this is a mistake, we indeed want the annotation as per [this comment](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3969#issuecomment-1420145285). We may have just not ported it to upgradeable contracts

Comment (frangio) [reply scripts/upgradeable/upgradeable.patch:145]: This comment is duplicate, discussing in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4424#discussion_r1253074417

**Comment by frangio** [comment]
Note: The concern about EIP712 was resolved in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4424#discussion_r1253074417

**Review by ernestognw — APPROVED** [review]
LGTM

**Pull request merged**